src/main/om/next/impl/parser.cljc

```diff
@@ -290,6 +290,6 @@
                              @mut-ret (assoc-in [key :result] @mut-ret)
                              @error (assoc key {:om.next/error @error}))))))))]
          (cond-> (reduce step (if (nil? target) {} []) query)
-           (not (or (not (nil? target)) elide-paths?)) (path-meta path query)))))))
+           (and (nil? target) (not elide-paths?)) (path-meta path query)))))))
```

Comment:
I took the liberty of simplifying this condition. nested `not`s were not very readable